docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+// options: some options
+// returns - ()
+pub fn indy_register_did_method(did_resolver_handle,
+                                method,
+                                ..callbacks,
+                                options) -> Future<()>
+```
+
+#### Close did resolver and releases all related resources
+
+```Rust
+// Close did resolver and releases all related resources.
+//
+// did_resolver_handle: resolver handle
+// returns - ()
+pub fn indy_close_did_resolver(did_resolver_handle) -> Future<()>
+```
+
+### Resolving DID Doc
+
+#### Resolve DID Doc for DID
+
+```Rust
+// Resolve DID Doc for DID.
+//
+// did_resolver_handle: resolver handle
+// did: did to resolve
+// options: some options, like cache behaviour
+// returns - did_doc: resolved DID Doc
+pub fn indy_resolve_did_doc(did_resolver_handle,
+                            did,
+                            options) -> Future<did_doc>
+```
+
+### DID and DID keys creation
+
+#### Creates a new DID
+
+```Rust
+// Create a new DID.
+//
+// Note: It doesn't create any records in the wallet.
+//
+// did_resolver_handle: resolver handle
+// method: did method to use
+// options: some options
+// returns - did: Generated DID
+pub fn indy_create_did(did_resolver_handle,
+                       method,
+                       options) -> Future<did>
+```
+
+#### Create a new key pair associated with DID
+
+```Rust
+// Create a new key pair associated with DID.
+//
+// It verifies that key options are compatible with method,
+// creates and stores key in the wallet, creates and stores association
+// of this key with did.
+//
+// Note: It can produce the same data as indy_create_my_did.
+//
+// did_resolver_handle: resolver handle
+// did: did to associate with this key
+// options: some options like key type and seed
+// returns - key_json: json that describes the key. Can be passed to did
+//                     doc builder
+pub fn indy_create_did_key(did_resolver_handle,
+                           did,
+                           options) -> Future<key_json>
+```
+
+### DID Doc creation, updating and publishing
+
+#### Create a new DID Doc builder instance
+
+```Rust
+// Create a new DID Doc builder instance.
+//
+// It tries to resolve did doc for provided did and creates builder that
+// allows to modify did doc and publish it. If did doc isn't exists it
+// starts building from scratch
+//
+// did_resolver_handle: resolver handle
+// did: did to associate did doc
+// options: some options like key type and seed
+// returns - ddoc_builder_handle: did doc builder handle
+pub fn indy_create_did_doc_builder(did_resolver_handle,
+                                   did,
+                                   options) -> Future<ddoc_builder_handle>
+```
+
+#### Add new public key item to DID Doc builder
+
+```Rust
+// Add new public key item to DID Doc builder.
+//
+// Note: id can be full DID Doc reference or fragment.
+// In case of fragment reference will be related to current DID.
```

Comment:
How is the current DID tracked? Is there a way to query such state management?

Reply:
I didn't get the question. The DID is one of the arguments for DID DOC Builder.